src/System.Linq/src/System/Linq/Range.cs

```diff
@@ -101,6 +101,18 @@ public List<int> ToList()
                 return list;
             }
 
+            public HashSet<int> ToHashSet(IEqualityComparer<int> comparer)
+            {
+                HashSet<int> hashSet = new HashSet<int>(_end - _start, comparer);
```

Comment:
I think it would be reasonable to call `Utilities.AreEqualityComparersEqual(comparer, EqualityComparer<int>.Default)` and pass the count when it's true. It would catch a common case where the optimisation is safe and the only potential downside is it would allocate `EqualityComparer<int>.Default` if it hadn't been used already but the odds that something already uses that particularly common use of `EqualityComparer` somewhere in the application is pretty high, making the downside minor.